### PR TITLE
Fusion: Add palette randomizer

### PR DESCRIPTION
**Pull request by JeffGainsNGames**
Adds basic palette randomizer options to cosmetics.
image
Room for future growth with hue_min, hue_max, symmetric options but is beyond the scope of this PR.

### CODE REVIEW

**Comment by Miepee** [comment]
Did you forget or plan to put the future settings (stereo, disable music(sfx) in a future PR?

**Comment by JeffGainsNGames** [comment]
Future PR. next task..

File: randovania/games/fusion/gui/dialog/cosmetic_patches_dialog.py [hunk 1]
```diff
@@ -1,38 +1,58 @@
...
-        # More signals here!
+
+        # Checkboxes for enabling Pallete Rando
+        self._persist_check_field(self.suit_palette_check, "enable_suit_palette")
+        self._persist_check_field(self.beam_palette_check, "enable_beam_palette")
+        self._persist_check_field(self.enemy_palette_check, "enable_enemy_palette")
+        self._persist_check_field(self.tileset_palette_check, "enable_tileset_palette")
+
+        # Combobox for Color Space
+        self.color_space_combo.currentIndexChanged.connect(self._on_color_space_update)
```
Comment (Miepee) [inline randovania/games/fusion/gui/dialog/cosmetic_patches_dialog.py:43]: why newline

File: test/games/fusion/gui/test_fusion_cosmetic_patches_dialog.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    dialog = FusionCosmeticPatchesDialog(None, cosmetic_patches)
+    skip_qtbot.addWidget(dialog)
+
+    skip_qtbot.mouseClick(dialog.beam_palette_check, QtCore.Qt.MouseButton.LeftButton)
+
+    assert dialog.cosmetic_patches == FusionCosmeticPatches(enable_beam_palette=True)
+
+
+def test_enemy_palette(skip_qtbot):
```
Comment (Miepee) [inline test/games/fusion/gui/test_fusion_cosmetic_patches_dialog.py:31]: Please combine all of these test checkbox functions.
You can just feed in parameters on whats different. Should only need the field, and the checkbox.

**Review by Miepee — APPROVED** [review]

**Pull request merged**